src/pages/library/datasetBuilder/DatasetBuilder.ts

```diff
@@ -0,0 +1,418 @@
+export const ValuesSelector = ({
+  selectedValues,
+  values,
+  onChange,
+}: {
+  selectedValues: HeaderAndValues<Value>[];
+  values: HeaderAndValues<Value>[];
+  onChange: (values: HeaderAndValues<Value>[]) => void;
+}) => {
+  return h(Selector, {
+    headerAction: div([
+      h(
+        LabeledCheckbox,
+        {
+          checked: values.length !== 0 && _.isEqual(values, selectedValues),
+          onChange: (checked) => (checked ? onChange(values) : onChange([])),
+          disabled: values.length === 0,
+        },
+        [label({ style: { paddingLeft: '0.5rem' } }, ['Select All'])]
+      ),
+    ]),
+    number: 3,
+    onChange,
+    datasetBuilderObjectSets: values,
+    selectedDatasetBuilderObjectSets: selectedValues,
+    header: 'Select values (columns)',
+    placeholder: div([
+      div(['No inputs selected']),
+      div(['You can view the available values by selecting at least one cohort and concept set']),
+    ]),
+    style: { width: '40%', marginLeft: '1rem' },
+  });
+};
+
+export const DatasetBuilderContents = ({ onStateChange }: { onStateChange: OnStateChangeType }) => {
+  const [selectedCohorts, setSelectedCohorts] = useState([] as HeaderAndValues<Cohort>[]);
+  const [selectedConceptSets, setSelectedConceptSets] = useState([] as HeaderAndValues<ConceptSet>[]);
+  const [selectedValues, setSelectedValues] = useState([] as HeaderAndValues<Value>[]);
+
+  return div({ style: { padding: `${PAGE_PADDING_HEIGHT}rem ${PAGE_PADDING_WIDTH}rem` } }, [
+    h2(['Datasets']),
+    div([
+      'Build a dataset by selecting the concept sets and values for one or more of your cohorts. Then export the completed dataset to Notebooks where you can perform your analysis',
+    ]),
+    ul({ style: { display: 'flex', width: '100%', marginTop: '2rem', listStyleType: 'none', padding: 0 } }, [
+      h(CohortSelector, {
+        selectedCohorts,
+        onChange: (cohorts) => {
+          setSelectedCohorts(cohorts);
+        },
+        onStateChange,
+      }),
+      h(ConceptSetSelector, {
+        selectedConceptSets,
+        onChange: (conceptSets) => setSelectedConceptSets(conceptSets),
+        onStateChange,
+      }),
+      h(ValuesSelector, { selectedValues, values: [], onChange: (values) => setSelectedValues(values) }),
+    ]),
+  ]);
+};
+
+type DatasetBuilderState = 'homepage' | 'cohort-editor' | 'concept-selector' | 'concept-set-creator';
+type OnStateChangeType = (state: DatasetBuilderState) => void;
+
+interface DatasetBuilderProps {
+  datasetId: string;
+}
+
+export const DatasetBuilderView = ({ datasetId }: DatasetBuilderProps) => {
+  const [datasetDetails, loadDatasetDetails] = useLoadedData<DatasetResponse>();
+  const [datasetBuilderState, setDatasetBuilderState] = useState<DatasetBuilderState>('homepage');
+
+  useOnMount(() => {
+    void loadDatasetDetails(() => DatasetBuilder().retrieveDataset(datasetId));
+  });
+  return datasetDetails.status === 'Ready'
+    ? h(FooterWrapper, {}, [
```

Comment:
Not sure if you got this working already, but I had to add `{ alwaysShow: true }` to my use of `Footerwrapper` to make it show up.